merlin/core/dispatch.py

```diff
@@ -47,9 +45,18 @@
             # cudf < 21.08
             from cudf.utils.dtypes import is_list_dtype as cudf_is_list_dtype
             from cudf.utils.dtypes import is_string_dtype as cudf_is_string_dtype
-
     except ImportError:
-        HAS_GPU = False
+        ...
```

Comment:
yes, there can be... Suppose your on a system that has cupy but does not have cudf. In the current method you find yourself not able to import cupy because cudf failed. And since we dont really require cudf for the use of Systems and this package is a base component for that. I dont think we should be grouping those packages all together. If a package is available it should be able to import.

Reply:
I guess I misunderstood your comment initially. I think you are highlighting that it doesn't need to be part of this PR. I can see that. I will make a separate PR for those changes. But the motivation behind them, I think, is still valid.